.github/workflows/main.yml

```
@@ -54,6 +54,9 @@ jobs:
       - name: Lint (line terminators)
         if: startsWith(matrix.os, 'ubuntu')
         run: "! grep 'with CRLF line terminators' <(git ls-files | file -nNf-)"
+      - name: Lint (unicode bidirectional control characters)
+        if: startsWith(matrix.os, 'ubuntu')
+        run: "! grep -E -- $'[\u2066-\u2069\u202A-\u202E]' <(git ls-files)"
```

Wait a sec, this checks file names, not file contents.

Is there a better way to fix this other than this:

```suggestion
        run: "! grep -E -- $'[\u2066-\u2069\u202A-\u202E]' $(git ls-files)"
```

This might exceed the argument length. I don't want to pipe grep into xargs because I want grep to be first.

---

Does this work?
```suggestion
        run: "! git grep -E -- $'[\u2066-\u2069\u202A-\u202E]'"
```

---

Git doesn't like character ranges for some reason. That should work, I think. Can you confirm before I update?

```sh
git grep -EIn $'[\u2066\u2067\u2068\u2069\u202A\u202B\u202C\u202D\u202E]'
```

---

I also couldn't get the ranges to work, I thought it might be some console option or user-error. 

Those characters seem to cover all those listed in the mozilla page. 👍

---

Hm, this confuses Github's output.
https://github.com/streamlink/streamlink/runs/4081108757?check_suite_focus=true#step:7:1

Is there a different way to write the unicode characters for the bash script, so that it doesn't mess with the log output?

---

Works:
https://github.com/streamlink/streamlink/runs/4081244407?check_suite_focus=true#step:7:2

---

LGTM